testenv/bootbox/web/genesis.rb

```diff
@@ -4,30 +4,38 @@
 require 'sinatra/base'
 
 module Genesis
-  class App < Sinatra::Base 
+  class App < Sinatra::Base
     configure do
       enable :logging, :dump_errors, :raise_errors
     end
 
+    def test_and_send_file file, *send_file_args
+      if file.size > 0
+        send_file file, *send_file_args
+      else
+        [404, 'file not found or empty']
+      end
+    end
+
     get '/ipxe-images/:file?' do
       file =  File.join('', 'genesis', 'bootcd', 'output', params[:file])
-      send_file file
+      test_and_send_file file
     end
 
     get '/gem/:name?' do
       # get the latest version gem of that name
       base =  File.join('', 'genesis', 'src' ,'*', params[:name])
       file = Dir.glob(base + '*.gem').sort.last
-      send_file file
+      test_and_send_file file
     end
 
     get '/testenv/:file?' do
       file =  File.join('', 'testenv', params[:file])
       if ['.yaml', '.yml'].include? File.extname(file)
-        send_file file, :type => 'application/x-yaml'
+        test_and_send_file file, :type => 'application/x-yaml'
       else
-        send_file file, :type => 'text/plain'
-      end 
+        test_and_send_file file, :type => 'text/plain'
```

Comment:
Does it bother you that this will block returning an empty file?  While we don't need to do so now, I can envision us wanting to have an empty config file or something.  This is probably fine, but ...